File: dfa.rb
# dfa.rb : finite state machine in Ruby
require_relative 'table'

i=ARGV[0]
s=0
until i.empty?
  c = i[0]; i = i[1..]
#puts i
  s = delta(s, c)
end
puts isaccept(s)
puts "s was: #{s}"

File: javascript/dfa.js
// deterministic finite automoton
const {  delta, isaccept } = require("./table");
var argv = process.argv;


var q = 0; // starting state number

for (char of argv[2]) {
  q = delta(q, char);
}
console.log("The value of q is: ", q);
console.log("The result is ", isaccept(q));